util/xivapi.ts

```diff
@@ -67,18 +71,60 @@ export class XivApi {
     if (columns.length === 0)
       exitError(`Cannot query API endpoint ${endpoint}: No columns specified.`);
 
-    const url = `${_XIVAPI_URL}${endpoint}?limit=${_XIVAPI_RESULTS_LIMIT}&columns=${
-      columns.join(',')
-    }`;
-
-    // TODO: Add some error detection & handling
-    const response = await fetch(url);
-    const json = (await response.json()) as XivApiResult;
+    let currentPage = 0;
+    let maxPage = 1;
+    let onePageJsonResults: XivApiResultData = {};
+    let useMultiPage = false;
+    const multiPageJson: { [page: number]: XivApiResult } = {};
+    while (currentPage < maxPage) {
+      currentPage++;
+      let url = `${_XIVAPI_URL}${endpoint}?limit=${_XIVAPI_RESULTS_LIMIT}&columns=${
+        columns.join(',')
+      }`;
+
+      if (currentPage !== 1)
+        url += `&page=${currentPage}`;
+      // TODO: add better error handling?
+      const response = await fetch(url);
+      const newPageJson = (await response.json()) as XivApiResult;
+
+      // first time through
+      if (currentPage === 1) {
+        maxPage = typeof newPageJson.Pagination.PageTotal === 'string'
+          ? parseInt(newPageJson.Pagination.PageTotal)
+          : newPageJson.Pagination.PageTotal;
+        if (maxPage === 1) {
+          onePageJsonResults = newPageJson.Results; // export as-is; we're done.
+          break;
+        }
+      }
+
+      // multiple pages of data so this gets a little kludgy.
+      // Response.json() will take the Results json, and for each record within, it will
+      // add that record object to (output).Results{} with an incrementing key.
+      // But if we have page 2+, we can't just dump into the same Results object because
+      // it will reuse the same range of incrementing keys, which will squash the existing data.
+      // So we need to keep each output.Results{} separate and merge at the end, manually
+      // numbering each (output).Results.(row) so there's no overlap.
+      useMultiPage = true;
+      multiPageJson[currentPage] = newPageJson;
```

Comment:
If nothing actually cares about the incrementing keys, why even keep them? This method could just return an array, and each page could just `.push(...values)` to the array?